### DISCUSSION TITLE
transmitter on ESP32 board

### DISCUSSION THREAD

**Discussion by maxotkin** [pschatzmann/arduino-audio-tools#1721] [Q&A] [answered]
Hello! I was directed to your library by ChatGPT. I’m not very knowledgeable about Bluetooth and sound conversion, as I specialize more in smart homes (Home Assistant). I have a 5.1 home audio system, but I can’t run wires to the rear speakers, so I wanted to transmit sound wirelessly. Using TWS Bluetooth modules and batteries, I converted the rear speakers to wireless. I bought several transmitters to send the sound to them, but the problem is that the first transmitter only transmitted mono sound, and the second one gave stereo sound but with a very large delay. ChatGPT suggested building a custom transmitter based on an ESP32 board and using your library. Is it possible to simply take an analog signal (from a standard 3.5mm jack) and send it via Bluetooth to the rear speakers? ChatGPT suggested the following code:

```
#include "BluetoothA2DPSource.h"
BluetoothA2DPSource a2dp_source;

// Pins for reading analog data
const int leftChannelPin = 34;  // Pin for the left channel
const int rightChannelPin = 35; // Pin for the right channel

// Callback for transmitting sound
int32_t get_sound_data(Frame *data, int32_t frameCount) {
    for (int i = 0; i < frameCount; i++) {
        int16_t leftChannelValue = analogRead(leftChannelPin);  // Reading the left channel
        int16_t rightChannelValue = analogRead(rightChannelPin); // Reading the right channel
        
        // Setting values for stereo
        data[i].channel1 = leftChannelValue;
        data[i].channel2 = rightChannelValue;
    }
    return frameCount;
}

void setup() {
  // Start Bluetooth with the device name and callback function
  a2dp_source.start("YMX-868BT", get_sound_data);
}

void loop() {
  // The main loop can remain empty since data is transmitted via the callback
}
```
I have serious doubts about the correctness of such signal conversion from analog to digital. It seems that ChatGPT made a significant mistake here.

If such conversion is not possible, could you suggest a few ideas on how to transmit sound wirelessly to the speakers? It doesn’t necessarily have to be Bluetooth. I’m willing to modify the speakers to use a different type of data reception. The sound quality doesn’t need to be high, as the rear speakers are mainly for creating surround sound and effects. In this case, the most important thing is low latency. Thank you in advance!

**Comment by pschatzmann** [comment, marked as answer]
Don't trust ChatGPT, it is ususlly just hallucinating solutions that won't work: the issue with the code is 
-  it does not sample at a defined audio sampling rate
- the values hava an inproper range

As a starrting point read the Wiki of this project to get an introduction into Audio and the hardware section.

The communications examples provide plenty of alternarives
Transmitting Bluetooth has quite some lag and I would not recommend it if you are a beginner since it is quite tricky and you can't use A2DP and WIFI toghether.
